### PR TITLE
Add user deposits and total deposits to getData

### PR DESCRIPTION
**Pull request by brettvniekerk**

### CODE REVIEW

**Review by nabsta-eth — APPROVED** [review]

**Review by nabsta-eth — DISMISSED** [review]

File: src/components/RewardPool.ts [hunk 1]
```diff
@@ -50,6 +53,16 @@ type RewardPoolIdsMulticall = Record<
...
+// type taken from rewardPool contract
+type Pool = [BigNumber, number, string, BigNumber, BigNumber, BigNumber] & {
+  weight: BigNumber;
+  assetType: number;
+  assetAddress: string;
+  totalDeposits: BigNumber;
+  S: BigNumber;
+  totalRealDeposits: BigNumber;
+};
+
```
Comment (Preston-Harrison) [inline src/components/RewardPool.ts:65]: if there is a way you could import this directly, that would be cool

File: src/components/RewardPool.ts [hunk 2]
```diff
@@ -84,10 +97,29 @@ export default class RewardPool {
...
+        provider
+      );
+      const totalDepositsMulticall = this.getTotalDepositsMulticall(
+        signer,
+        rewardPoolIds
+      );
+      const totalDepositsResponse = await callMulticallObject(
+        totalDepositsMulticall,
+        provider
+      );
```
Comment (Preston-Harrison) [inline src/components/RewardPool.ts:116]: use Promise.all as these don't look dependant on each other

**Review by Preston-Harrison — CHANGES_REQUESTED** [review]

File: src/components/RewardPool.ts (hunk 1 above)
Comment (brettvniekerk) [reply src/components/RewardPool.ts:65]: it is not exported so i cant unfortunately

File: src/components/RewardPool.ts (hunk 2 above)
Comment (brettvniekerk) [reply src/components/RewardPool.ts:116]: 👍

**Review by Preston-Harrison — APPROVED** [review]

File: src/components/RewardPool.ts [hunk 3]
```diff
@@ -196,6 +222,43 @@ export default class RewardPool {
...
+    return Object.keys(pools).reduce(
+      (prevObject, key) => ({
+        ...prevObject,
+        [key]: contracts.rewardPool.getDeposit(account, pools[key]),
+      }),
+      {}
+    );
+  };
+
+  private getTotalDepositsMulticall = (
```
Comment (rellfy) [inline src/components/RewardPool.ts:244]: This is getting the pools, not total deposits specifically. Could you rename the function?

Comment (brettvniekerk) [reply src/components/RewardPool.ts:244]: will do

**Review by Preston-Harrison — DISMISSED** [review]

**Review by rellfy — APPROVED** [review]

**Review by nabsta-eth — APPROVED** [review]

**Pull request merged**